### PR TITLE
stabilityai-stable-diffusion-xl-base-1-0

### PR DESCRIPTION
**Pull request by Suvratpalt**
Job Link: https://ml.azure.com/experiments/id/50152fe1-d4fa-41e1-976f-c5125387e209/runs/happy_clock_s84pvsj2q6?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rv-images-rc/providers/Microsoft.MachineLearningServices/workspaces/rv-images-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

batch job link: https://ml.azure.com/runs/batchjob-830beb40-1291-46fc-b2a4-fc87c1a5d927?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourcegroups/rv-images-rc/workspaces/rv-images-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

**conda.yaml**
```
channels:
- anaconda
- pytorch
- conda-forge
dependencies:
- python=3.8.16
- pip<=23.0.1
- pip:
  - mlflow==2.6.0
  - torch==1.13.1
  - transformers==4.33.2
  - diffusers==0.23.0
  - accelerate==0.22.0
  - azure-ai-mlmonitoring
  - azure-identity==1.13.0
  - azure-mgmt-cognitiveservices==13.4.0
  - azure-ai-contentsafety==1.0.0b1
  - aiolimiter
name: mlflow-env
```
**MLmodel**
```
flavors:
  python_function:
    artifacts:
      model_dir:
        path: artifacts/INPUT_model_path
        uri: /mnt/azureml/cr/j/c820a7fdbbe948ffb696f92545b13a6b/cap/data-capability/wd/INPUT_model_path
    cloudpickle_version: 2.2.1
    code: code
    env:
      conda: conda.yaml
      virtualenv: python_env.yaml
    loader_module: mlflow.pyfunc.model
    python_model: python_model.pkl
    python_version: 3.8.18
metadata:
  base_model_name: stabilityai/stable-diffusion-xl-base-1.0
  base_model_task: text-to-image
  is_acft_model: true
  is_finetuned_model: false
  model_type: stable-diffusion
mlflow_version: 2.8.1
model_size_bytes: 76912778398
model_uuid: 89e94e78019f456e97e26dce1ced948a
signature:
  inputs: '[{"type": "string", "name": "prompt"}]'
  outputs: '[{"type": "string", "name": "prompt"}, {"type": "binary", "name": "generated_image"},
    {"type": "boolean", "name": "nsfw_content_detected"}]'
  params: null
utc_time_created: '2024-02-01 08:39:51.458723'
```

### CODE REVIEW

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+> Note:
+>
+> - "image" string is in base64 format.
+> - The `stabilityai-stable-diffusion-xl-base-1.0` model doesn't check for the NSFW content in generated image. We highly recommend to use the model with <a href="https://learn.microsoft.com/en-us/azure/ai-services/content-safety/studio-quickstart" target="_blank">Azure AI Content Safety (AACS)</a>. Please refer sample <a href="https://aka.ms/azureml-infer-sdk-safe-text-to-image" target="_blank">online</a>  and <a href="https://aka.ms/azureml-infer-batch-sdk-safe-text-to-image" target="_blank">batch</a> notebooks for AACS integrated deployments.
+
+#### Visualization for the prompt - "a photograph of an astronaut riding a horse"
+
+<img src="https://automlcesdkdataresources.blob.core.windows.net/finetuning-image-models/images/Model_Result_Visualizations(Do_not_delete)/output_stabilityai_stable_diffusion_2_1.png" alt="stabilityai_stable_diffusion_2_1 visualization">
```
Comment (rjaincc) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md:88]: Please use the right viz image. Not the viz image from stable diffusion 2.1

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+properties:
+  SharedComputeCapacityEnabled: true
+  SHA: 462165984030d82259a11f4367a4eed129e94a7b
+  inference-min-sku-spec: 6|1|112|336
+  inference-recommended-sku: Standard_NC6s_v3, Standard_NC12s_v3, Standard_NC24s_v3,
+    Standard_ND40rs_v2, Standard_ND96amsr_A100_v4, Standard_ND96asr_v4
+tags:
+  Preview: ""
+  SharedComputeCapacityEnabled: ""
+  license: creativeml-openrail++-m
```
Comment (rjaincc) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml:13]: @vadthyavath - Please check the license etc.

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/model.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+path:
+  container_name: finetuning-image-models
+  container_path: HuggingFace-mlflow/stabilityai-stable-diffusion-xl-base-1-0/1707841523/mlflow_model_folder
```
Comment (vadthyavath) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/model.yaml:3]: 1707841523? can u verify this? I don't see that directory in datastore

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md (hunk 1 above)
Comment (vadthyavath) [reply assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md:88]: please generate the sample image using the same input from stable diffusion 2.1 and use that here

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  task: text-to-image
+  huggingface_model_id: stabilityai/stable-diffusion-xl-base-1.0
+  author: stabilityai
+  training_dataset: LAION-5B
+  inference_supported_envs:
+  - ds-mii
```
Comment (vadthyavath) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml:19]: we need to verify ds-mii. please confirm with Gaurav

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  inference_compute_allow_list:
+    [
+      Standard_NC6s_v3,
+      Standard_NC12s_v3,
+      Standard_NC24s_v3,
+      Standard_ND40rs_v2,
+      Standard_ND96amsr_A100_v4,
+      Standard_ND96asr_v4
+    ]
+version: 9
```
Comment (vadthyavath) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml:29]: we r releasing first version not 9

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
+This `stabilityai-stable-diffusion-xl-base-1.0` model is fine-tuned from [stable-diffusion-xl-base-1.0](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0) 
+
+The SD-XL 1.0-base Model, developed by Stability AI, is a Diffusion-based text-to-image generative model. It utilizes a two-stage pipeline for latent diffusion, consisting of a base model and a refinement model. The base model generates noisy latents, which are then processed by a refinement model for final denoising. Alternatively, a two-stage pipeline involves using a specialized high-resolution model and applying the SDEdit technique to the latents generated in the first step.
```
Comment (vadthyavath) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md:3]: where did u get this content from?

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+The model is available on GitHub, and for research purposes, Stability AI recommends their generative-models repository. The model is licensed under the CreativeML Open RAIL++-M License. It is designed for various research areas, such as generating artworks, educational tools, and understanding the limitations and biases of generative models. However, it is not intended for factual or true representations of people or events.
+
+The model has limitations, including not achieving perfect photorealism, struggling with tasks involving compositionality, and facing challenges with rendering faces and legible text. The autoencoding part of the model is lossy, and there is a potential for reinforcing social biases in generated content. The model's deployment is restricted to safe applications, and certain use cases, like generating factual content, are considered out-of-scope for its abilities.
+
+You can find more examples in optimum https://huggingface.co/docs/optimum/main/en/onnxruntime/usage_guides/models#stable-diffusion-xl
```
Comment (vadthyavath) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md:9]: remove thi spart on optimum

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: stabilityai-stable-diffusion-xl-base-1.0
```
Comment (rjaincc) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml:2]:
```suggestion
name: stabilityai-stable-diffusion-xl-base-1-0
```

Comment (rjaincc) [reply assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml:2]: Let's follow same naming convention across the board..
For example, we have "Salesforce-BLIP-2-opt-2-7b-image-to-text"

Comment (Suvratpalt) [reply assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml]: Changed the Directory name Thanks

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md (hunk 1 above)
Comment (vadthyavath) [reply assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/description.md]: added!

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/model.yaml (hunk 3 above)
Comment (vadthyavath) [reply assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/model.yaml]: updated

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml (hunk 4 above)
Comment (vadthyavath) [reply assets/models/system/stabilityai-stable-diffusion-xl-base-1.0/spec.yaml]: ds-mii is not working due to following 2 issues:
1. prod image for deepspeed is not accessible due to some issue
2. when we tested against the local docker image on workspace, we observed that the inference code is different for ds-mii and it doesn't validate teh nsfw_content_detected attribute which is not available for xl pipelines from HF

will address separatel in next round!

File: assets/models/system/stabilityai-stable-diffusion-xl-base-1-0/spec.yaml [hunk 9]
```diff
@@ -0,0 +1,26 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: stabilityai-stable-diffusion-xl-base-1.0
```
Comment (rjaincc) [inline assets/models/system/stabilityai-stable-diffusion-xl-base-1-0/spec.yaml:2]:
```suggestion
name: stabilityai-stable-diffusion-xl-base-1-0
```

**Review by rjaincc — APPROVED** [review]

**Pull request merged**